packages/tpetra/core/src/Tpetra_Details_crsUtils.hpp

```diff
@@ -229,6 +230,14 @@ pad_crs_arrays(
     // DEEP_COPY REVIEW - DEVICE-TO-HOSTMIRROR
     Kokkos::deep_copy(execution_space(), row_ptr_beg_h, row_ptr_beg);
 
+    // lbv 03/15/23: The execution space deep_copy does an asynchronous
+    // copy so we really want to fence that space before touching the
+    // host data as it is not guarenteed to have arrived by the time we
+    // start the parallel_reduce below which might use a different
+    // execution space, see:
+    // https://kokkos.github.io/kokkos-core-wiki/API/core/view/deep_copy.html#semantics
+    exec_space_instance.fence();
```

Comment:
Note the reason why it may or may not cause issues on other backends is that on some backends allocations are blocking. But that is not guaranteed, so this fence needs to be here.